linode_api4/objects/linode_interfaces.py

```diff
@@ -0,0 +1,414 @@
+    """
+
+    range: str = ""
+    route_target: Optional[str] = None
+
+
+@dataclass
+class LinodeInterfacePublicIPv6Range(JSONObject):
+    """
+    A single range under the IPv6 configuration of a public Linode Interface.
+    """
+
+    put_class = LinodeInterfacePublicIPv6RangeOptions
+
+    range: str = ""
+    route_target: Optional[str] = None
+
+
+@dataclass
+class LinodeInterfacePublicIPv6(JSONObject):
+    """
+    The IPv6 configuration of a Linode Interface.
+    """
+
+    put_class = LinodeInterfacePublicIPv6Options
+
+    slaac: List[LinodeInterfacePublicIPv6SLAAC] = field(default_factory=list)
+    shared: List[LinodeInterfacePublicIPv6Shared] = field(default_factory=list)
+    ranges: List[LinodeInterfacePublicIPv6Range] = field(default_factory=list)
+
+
+@dataclass
+class LinodeInterfacePublic(JSONObject):
+    """
+    Public-specific configuration fields for a Linode Interface.
+    """
+
+    put_class = LinodeInterfacePublicOptions
+
+    ipv4: Optional[LinodeInterfacePublicIPv4] = None
+    ipv6: Optional[LinodeInterfacePublicIPv6] = None
+
+
+@dataclass
+class LinodeInterfaceVLAN(JSONObject):
+    """
+    VLAN-specific configuration fields for a Linode Interface.
+    """
+
+    put_class = LinodeInterfaceVLANOptions
+
+    vlan_label: str = ""
+    ipam_address: Optional[str] = None
+
+
+class LinodeInterface(DerivedBase):
+    """
+    A Linode's network interface.
+
+    NOTE: When using the ``save()`` method, certain local fields with computed values will
+          not be refreshed on the local object until after ``invalidate()`` has been called::
+
+            # Automatically assign an IPv4 address from the associated VPC Subnet
+            interface.vpc.ipv4.addresses[0].address = "auto"
+
+            # Save the interface
+            interface.save()
+
+            # Invalidate the interface
+            interface.invalidate()
+
+            # Access the new address
+            print(interface.vpc.ipv4.addresses[0].address)
+
+    API Documentation: https://techdocs.akamai.com/linode-api/reference/get-linode-interface
+    """
+
+    api_endpoint = "/linode/instances/{linode_id}/interfaces/{id}"
+    derived_url_path = "interfaces"
+    parent_id_name = "linode_id"
+
+    properties = {
+        "linode_id": Property(identifier=True),
+        "id": Property(identifier=True),
+        "mac_address": Property(),
+        "created": Property(is_datetime=True),
+        "updated": Property(is_datetime=True),
+        "version": Property(),
+        "default_route": Property(
+            mutable=True,
+            json_object=LinodeInterfaceDefaultRoute,
+        ),
+        "public": Property(mutable=True, json_object=LinodeInterfacePublic),
+        "vlan": Property(mutable=True, json_object=LinodeInterfaceVLAN),
+        "vpc": Property(mutable=True, json_object=LinodeInterfaceVPC),
+    }
+
+    def get_firewalls(self, *filters) -> List[Firewall]:
```

Comment:
I used the name `get_firewalls(...)` rather than `firewalls(...)` here to prevent ambiguity between this and similar property methods. 

Any thoughts on this? I'd be happy to revert it if we feel it's too out of line with the rest of the SDK 🙂